run.sh

```diff
@@ -5,8 +5,22 @@ then
     gh api /repos/$GITHUB_REPOSITORY/releases | jq -e -r '.[] | select(.draft == true and .name == "next") | .body' | egrep "$INTERESTING_CATEGORIES"
 fi
 export MAVEN_OPTS=-Djansi.force=true
-mvn -B -V -s $GITHUB_ACTION_PATH/settings.xml -ntp -Dstyle.color=always -Dset.changelist -DaltDeploymentRepository=maven.jenkins-ci.org::default::https://repo.jenkins-ci.org/releases/ -Pquick-build -P\!consume-incrementals clean deploy
-version=$(mvn -B -ntp -Dset.changelist -Dexpression=project.version -q -DforceStdout help:evaluate)
-gh api -F ref=refs/tags/$version -F sha=$GITHUB_SHA /repos/$GITHUB_REPOSITORY/git/refs
-release=$(gh api /repos/$GITHUB_REPOSITORY/releases | jq -e -r '.[] | select(.draft == true and .name == "next") | .id')
-gh api -X PATCH -F draft=false -F name=$version -F tag_name=$version /repos/$GITHUB_REPOSITORY/releases/$release
+if fgrep -sq changelist.format .mvn/maven.config
+then # JEP-229
+    mvn -B -V -s $GITHUB_ACTION_PATH/settings.xml -ntp -Dstyle.color=always -Dset.changelist -DaltDeploymentRepository=maven.jenkins-ci.org::default::https://repo.jenkins-ci.org/releases/ -Pquick-build -P\!consume-incrementals clean deploy
+    version=$(mvn -B -ntp -Dset.changelist -Dexpression=project.version -q -DforceStdout help:evaluate)
+    gh api -F ref=refs/tags/$version -F sha=$GITHUB_SHA /repos/$GITHUB_REPOSITORY/git/refs
+    name=next
+    tag=$version
+else # MRP
+    git config --global user.email cd@jenkins.io
+    git config --global user.name jenkins-maven-cd-action
+    git config --global url.https://github.com/.insteadOf git@github.com:
+    mvn -B -V -s $GITHUB_ACTION_PATH/settings.xml -ntp -Dstyle.color=always -P\!consume-incrementals -Darguments='-Pquick-build -ntp' validate release:prepare release:perform
```

Comment:
Of you are running a quick build you should probably just run validate as the preparation goal?

Reply:
The preparation goals verify that the artifact can actually be packaged. We also run `validate` first just to enforce the `-P!consume-incrementals`, which would not work inside preparation goals because of profile mayhem; https://github.com/jenkinsci/plugin-pom/pull/416 would also catch mistakes but only after junk commits & tag were pushed.

Reply:
I believe the action would have issues with branch protection 😢

Don't see this ever working unless you go around each repository and allow GitHub action bot to push commits past branch protections.

Reply:
See: https://github.com/jenkins-infra/jenkins-maven-cd-action/pull/14#issuecomment-985052932

Reply:
Maybe this gha could be used to try to pass over branch protections?
https://github.com/marketplace/actions/branch-protection-bot

Reply:
https://github.com/marketplace/actions/branch-protection-bot makes it doable.

Especially since we introduced https://github.com/jenkins-infra/github-reusable-workflows

Reply:
Although the GitHub action only removes include administrators, it does not remove branch protection entirely with status checks.

I believe you could create an GitHub action to retrieve current branch protect.
Remove the branch protection and restore it using an GitHub action.

Reply:
IIUC https://github.com/marketplace/actions/branch-protection-bot#access_token cannot use `$GITHUB_TOKEN`, which I think makes that a non-starter.